lmcache/storage_backend/local_backend.py

```diff
@@ -311,38 +353,105 @@ def remove(
 
     @_lmcache_nvtx_annotate
     def put_worker(self, ):
-        put_stream = torch.cuda.Stream()
         while True:
             item = self.put_queue.get()
             if isinstance(item, LocalBackendEndSignal):
                 break
             key, value = item
-            with torch.cuda.stream(put_stream):
-                self.put_blocking(key, value)
+            self.put_nonblocking(key, value)
 
+    @_lmcache_nvtx_annotate
+    @torch.inference_mode()
+    def save_disk(
+        self,
+        path: str,
+        kv_chunk: torch.Tensor,
+    ):
+        save_file({"kv_chunk": kv_chunk.contiguous()}, path)
+
+    @_lmcache_nvtx_annotate
+    @torch.inference_mode()
+    def put_nonblocking(
```

Comment:
I like the idea of using multi-process to save the cache. In this case, I also feel like we can get rid of the thread during the put.